### PR TITLE
Active active docs mention plugin support

### PR DESCRIPTION
**Pull request by johanandren**
Noticed we had nothing about this.

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/persistence-active-active.md [hunk 1]
```diff
@@ -283,4 +285,12 @@ and then enable direct replication through `withDirectReplication()` on @apidoc[
...
\ No newline at end of file
+to fast forward the stream of events for the origin replica. (With additional potential future support in journals for fast forwarding [#29311](https://github.com/akka/akka/issues/29311)). 
+
+## Journal Support
+
+For a journal plugin to support active active it needs to store and read metadata for each event if it is defined in the @apiref[PersistentRepr]
+ `metadata` field. To attach the metadata after writing it, `PersistentRepr.withMetadata` is used.
+ 
+For a snapshot plugin to support active active it needs to store and read metadata for the snapshot if it is defined in the @apiref[akka.persistence.SnapshotMetadata] `meta` field. 
+To attach the metadata when reading the snapshot the `akka.persistence.SnapshotMetadata.apply` factory overload taking a `meta` parameter is used.
```
Comment (renatocaval) [inline akka-docs/src/main/paradox/typed/persistence-active-active.md:296]: Is that correct that on one side we call it `metadata` and on the other side it's `meta`? Shouldn't we take the opportunity to align the naming?

**Review by octonato — APPROVED** [review]
LGTM

If we want to align the names (see comment), better to do in this PR.

File: akka-docs/src/main/paradox/typed/persistence-active-active.md (hunk 1 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/typed/persistence-active-active.md:296]: I agree, I'll rename it right away in this pr

File: akka-persistence/src/main/scala/akka/persistence/SnapshotProtocol.scala [hunk 2]
```diff
@@ -11,14 +11,14 @@ import scala.runtime.AbstractFunction3
...
- * @param meta a journal can optionally support persisting metadata separate to the domain state
+ * @param metadata a journal can optionally support persisting metadata separate to the domain state, used for active active support
  */
 @SerialVersionUID(1L)
 final class SnapshotMetadata(
     val persistenceId: String,
     val sequenceNr: Long,
     val timestamp: Long,
-    val meta: Option[Any])
+    val metadata: Option[Any])
```
Comment (renatocaval) [inline akka-persistence/src/main/scala/akka/persistence/SnapshotProtocol.scala:21]: I had this on a tab and forgot to send the comment. 

Now I understand why it had a different name. We have already a metadata and now we are adding an optional field to it. Strange to have `metadata.metadata`. 

I don't now what would make it more clear. 

* `extras` - `metadata.extras`
* `optional` - `metadata.optional`
* `modelMetadata` - `metadata.modelMetadata`
* `replica` - `metadata.replica`

We can also leave like that though.

**Pull request merged**